jobclient/src/main/java/com/twosigma/cook/jobclient/JobClient.java

```diff
@@ -1113,4 +1103,76 @@ public String toString() {
                 + ", _activeUUIDToJob=" + _activeUUIDToJob + ", _batchSubmissionLimit=" + _batchRequestSize
                 + ", _statusUpdateInterval=" + _statusUpdateInterval + "]";
     }
+
+    /**
+     * Create a {@link JobClientProxy} for submitting requests on behalf of another user.
+     * @param impersonatedUser specifies the authentication principal of the user to be impersonated.
+     * @return a {@link JobClientProxy} that can perform impersonated requests.
+     */
+    public JobClientProxy impersonating(String impersonatedUser) {
```

Comment:
Why can't this return a `JobClientInterface`?

Reply:
It could, but I wanted to emphasize the fact that this is just a *proxy* over the original JobClient; i.e., all of the mutable internal data structures are shared between the original JobClient object and the impersonation-proxy.